src/main/java/io/opentracing/contrib/global/GlobalTracer.java

```diff
@@ -0,0 +1,171 @@
+     * the {@link Tracer} service implementation.<br>
+     * If zero or more than one service implementations are found,
+     * the {@link io.opentracing.NoopTracer NoopTracer} will be returned.
+     *
+     * @return The non-<code>null</code> global tracer to use.
+     */
+    public static Tracer tracer() {
+        return INSTANCE.getOrInitTracer();
+    }
+
+    @Override
+    public SpanBuilder buildSpan(String operationName) {
+        return getOrInitTracer().buildSpan(operationName);
+    }
+
+    @Override
+    public <C> void inject(SpanContext spanContext, Format<C> format, C carrier) {
+        getOrInitTracer().inject(spanContext, format, carrier);
+    }
+
+    @Override
+    public <C> SpanContext extract(Format<C> format, C carrier) {
+        return getOrInitTracer().extract(format, carrier);
+    }
+
+    /**
+     * Wraps the {@link Callable} to execute within a new {@link Span} if an
+     * {@link TracedCallable#withOperationName(String) operationName} is also specified.<br>
+     * If no operationName is provided, the callable will execute as-is without starting a new span.
+     *
+     * @param callable The callable to wrap.
+     * @param <V>      The return type of the wrapped call.
+     * @return The wrapped call.
+     * @see TracedCallable#withOperationName(String)
+     */
+    public static <V> TracedCallable<V> traced(Callable<V> callable) {
+        return TracedCallable.of(callable);
+    }
+
+    /**
+     * Wraps the {@link Runnable} to execute within a new {@link Span} if an
+     * {@link TracedCallable#withOperationName(String) operationName} is also specified.<br>
+     * If no operationName is provided, the callable will execute as-is without starting a new span.
+     *
+     * @param runnable The runnable to wrap.
+     * @return The wrapped call.
+     * @see TracedRunnable#withOperationName(String)
+     */
+    public static TracedRunnable traced(Runnable runnable) {
+        return TracedRunnable.of(runnable);
+    }
+
+    /**
+     * Loads a single service implementation from {@link ServiceLoader}.
+     *
+     * @return The single service or a NoopTracer.
+     */
+    private static Tracer loadSingleton() {
+        Tracer foundSingleton = null;
+        for (Iterator<Tracer> implementations =
+             ServiceLoader.load(Tracer.class, Tracer.class.getClassLoader()).iterator();
+             foundSingleton == null && implementations.hasNext(); ) {
+            final Tracer implementation = implementations.next();
+            if (implementation != null) {
+                LOGGER.log(Level.FINEST, "Tracer service loaded: {0}.", implementation);
+                if (implementations.hasNext()) { // Don't actually load the next implementation, fall-back to default.
+                    LOGGER.log(Level.WARNING,
+                            "More than one Tracer service implementation found. Falling back to default implementation.");
```

Comment:
hmm it is worse than it appears. I don't think someone reading logs w/o knowledge of this library would guess default is noop. should noop be in the log message?